packages/rum/src/domain/segmentCollection/segment.ts

```diff
@@ -26,12 +29,19 @@ export class Segment {
     this.recordsCount = 1
     this.hasFullSnapshot = initialRecord.type === RecordType.FullSnapshot
 
+    this.viewStats = getOrCreateViewStats(context.view.id)
+    this.viewStats.segments_count += 1
+    this.viewStats.records_count += 1
```

Comment:
Could we move the computation in view stats file?

Reply:
What are you thinking about? Instead of having a `getOrCreateViewStats`, three functions like `incrementSegmentsCount(viewId)`, `incrementRecordsCount(viewId)` and `incrementRawSize(viewId)` ?

Reply:
Feels weird to have function like addSegment(viewId) which does not really add a segment. Wouldn't it make more sense to mimic the RUM logic and have a `replayLifeCycle` with `segmentCreated`, `recordCreated` and given that object the `viewStats` are computed?
Plus, it's easily extendable for future similar use cases.